src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/SpeedScopeStacksFormatter.cs

```diff
@@ -0,0 +1,127 @@
+                }
+            };
+
+            NameCache cache = stackResult.NameCache;
+
+            foreach (CallStack stack in stackResult.Stacks)
+            {
+                Models.Profile profile = new Profile()
+                {
+                    Events = new List<ProfileEvent>(),
+                    StartValue = FixedStart,
+                    EndValue = FixedEnd,
+                    Unit = UnitType.none,
+                    Name = string.Format(CultureInfo.CurrentCulture, Strings.CallstackThreadHeader, stack.ThreadId),
+                };
+
+                speedscopeResult.Profiles.Add(profile);
+
+                foreach (CallStackFrame frame in stack.Frames)
+                {
+                    if (frame.FunctionId == 0)
+                    {
+                        profile.Events.Add(new ProfileEvent { At = FixedStart, Frame = 0, Type = ProfileEventType.O });
+
+                    }
+                    else if (cache.FunctionData.TryGetValue(frame.FunctionId, out FunctionData functionData))
+                    {
+                        var builder = new StringBuilder();
+                        builder.Append(GetModuleName(cache, functionData.ModuleId));
+                        builder.Append(ModuleSeparator);
+                        BuildClassName(builder, cache, functionData);
+                        builder.Append(ClassSeparator);
+                        builder.Append(functionData.Name);
+                        BuildGenericParameters(builder, cache, functionData.TypeArgs);
+
+                        if (!functionToSharedFrameMap.TryGetValue(frame.FunctionId, out int mapping))
+                        {
+                            speedscopeResult.Shared.Frames.Add(new SharedFrame { Name = builder.ToString() });
+                            mapping = speedscopeResult.Shared.Frames.Count - 1;
+                            functionToSharedFrameMap.Add(frame.FunctionId, mapping);
+                        }
+
+                        var profileEvent = new ProfileEvent
+                        {
+                            At = FixedStart,
+                            Frame = mapping,
+                            Type = ProfileEventType.O,
+                        };
+
+                        profile.Events.Add(profileEvent);
+
+                    }
+                    else
+                    {
+                        profile.Events.Add(new ProfileEvent { At = FixedStart, Frame = 1, Type = ProfileEventType.O });
+                    }
+                }
+
+                // Speedscope requires a close event for each open event.
+                for (int i = profile.Events.Count - 1; i >= 0; i--)
+                {
+                    profile.Events.Add(new ProfileEvent { At = FixedEnd, Frame = profile.Events[i].Frame, Type = ProfileEventType.C });
+                }
+            }
+
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+#if NET6_0_OR_GREATER
```

Comment:
This shouldn't be necessary anymore since this should only be building for .NET 6+.